pkg/kube/apisix/apis/config/v1/types.go

```diff
@@ -126,6 +126,14 @@ type ApisixUpstreamConfig struct {
 	// service versions.
 	// +optional
 	Subsets []ApisixUpstreamSubset `json:"subsets,omitempty" yaml:"subsets,omitempty"`
+
+	// The host option sent to the upstream, value can be pass, node and rewrite.
```

Comment:
> That's a good point, there's another scenario the link is https://gist.github.com/elvis-cai/c3774867af113f901e69bde7e41de154, if using `curl -H "Host: local.httpbin.org" ${APISIX_GATEWAY} `, since by default pass_host is pass, apisix will use `host header: local.httpbin.org` to hit the ip address of `apple.internal.com`, or we will set `pass_host: node` by default when accessing ExternalName service in k8s?
> 
> Any suggestions for the naming `pass` and `rewrite` would be apprecated, thanks.

The literals should keep the "what you see is what you get" principle, so what about `"httpHostRewritePolicy"`, the candidates can be `"useOriginalClientHost"` and `"rewriteTo"` (in such a case, we have to use another field to specify the target host, saying `"target"`), and `useEndpointHostnameOrAddress`.

```yaml
httpHostRewritePolicy:
  kind: useOriginalClientHost

httpHostRewritePolicy:
  kind: rewriteTo
  target: blah.blah.foo.com

httpHostRewritePolicy:
  kind: useEndpointHostnameOrAddress
```

Reply:
applogise for the late reply, updated, here's another example of httpHostRewritePolicy https://gist.github.com/elvis-cai/1e7179f114bb4f8390cf6a63e8dc5fe4

Reply:
👍